lib/logstash/inputs/http_poller.rb

```diff
@@ -5,9 +5,18 @@
 require "socket" # for Socket.gethostname
 require "manticore"
 require "rufus/scheduler"
+require "logstash/plugin_mixins/ecs_compatibility_support"
+require 'logstash/plugin_mixins/ecs_compatibility_support/target_check'
+require 'logstash/plugin_mixins/validator_support/field_reference_validation_adapter'
+require 'logstash/plugin_mixins/event_support/event_factory_adapter'
 
 class LogStash::Inputs::HTTP_Poller < LogStash::Inputs::Base
   include LogStash::PluginMixins::HttpClient
+  include LogStash::PluginMixins::ECSCompatibilitySupport(:disabled, :v1, :v8 => :v1)
+  include LogStash::PluginMixins::ECSCompatibilitySupport::TargetCheck
```

Comment:
As you brought up in the weekly sync, the `target` here in the input plugin combined with the codec's `target` makes things awkward, especially with regard to how the JSON codec attempts to set an `event.original`. I think users should use one or the other -- but not both -- and that our messaging should help them along that path instead of warning them here when they are correctly only using the codec's `target`. 

I've looked pretty deeply into seeing if there were a way to propagate the "target" option of this plugin to the codec if the codec supports it, and it's a pretty deep rabbit hole. Every option I have pursued so far has really rough edges and I don't immediately see a path to implementing something like this that can work in a backward-compatible way. 

Instead of using the `TargetCheck` helper, I think we may need to do a different validation.

~~~ ruby
    if @codec.respond_to?(:target)
      if @target
        if @codec.target
          # WARN: targeting both is not good.
        else
          # WARN: target codec, not input
          line = "`codec => #{@codec.config_name} { target => #{@target.inspect} }`"
        end
      end
    elsif @target.nil? && ecs_compatibility != :disabled
      # WARN: likely to clash without target set; ignore if your inbound events are already ECS.
    end
~~~

I have the sense that this will play out in other inputs, too. It may be worth pulling this type of validation up into a sister module to the shared `TargetCheck`.